### DISCUSSION TITLE
Can Someone Show me an ExcelAsyncUtil.Observe example by C#?

### DISCUSSION THREAD

**Discussion by MasterBran** [Excel-DNA/ExcelDna#457] [Q&A] [answered]
I use an async Task function in UDF like this, now I use the `ExcelAsyncUtil.Run` to run that but the Excel process will keep alive even if i closed the Excel application.So I want to try `ExcelAsyncUtil.Observe` but i fund no C# example, please show me one example or any other solution to fix it.
`public static object GETROOT(object p0, object p1)
        {
            return ExcelAsyncUtil.Run("GETROOT", new object[] { p0, p1 }, delegate
            {
                return  UDFGenerator.GetApiResult("001659ca6e3ed001d9a9be59d903a15c", "2", "NM,ENM", new object[]{ p0, p1 }, ExcelApp).GetAwaiter().GetResult();
            });
        }`
PS: GetApiResult return a object[,] type value;

**Comment by govert** [comment, marked as answer]
I'm worried when I see `ExcelApp` inside the async method. I'm guessing this means you are using the Excel COM object model from the task which will run on a separate thread. You should never use the Excel COM object model form a worker thread. If you do, various problems can occur, including the Excel process not shutting down properly.

**Reply by MasterBran** [reply]
You were right, I will try other way to do that, thank you very much
